cmake/dependencies.cmake

```diff
@@ -1,8 +1,26 @@
-# >= 3.2 required for ExternalProject_Add_StepDependencies
-cmake_minimum_required(VERSION 3.2)
-
-
-include(${PROJECT_SOURCE_DIR}/lib/libappimage/cmake/scripts.cmake)
+# FetchContent_MakeAvailable() is only available in CMake 3.14 or newer
+cmake_minimum_required(VERSION 3.14)
+
+include(FetchContent)
+
+# Need this patch until https://github.com/AppImage/libappimage/pull/160 is resolved
+FetchContent_Declare(libappimage_patch
+    URL https://github.com/AppImage/libappimage/commit/b3398bb496e47947864b4b8bc2999c8427f86a9a.patch
+    DOWNLOAD_NO_EXTRACT TRUE
+)
+FetchContent_MakeAvailable(libappimage_patch)
+
+FetchContent_Declare(libappimage
+    # We can not use a URL source with a github-generated source archive: libappimage's gtest submodule would be missing
```

Comment:
Eventually we may just want to use gtest from the system (generally provided externally), these submodules are more than annoying anyway.